manager/logbroker/broker.go

```diff
@@ -28,24 +34,51 @@ type LogBroker struct {
 
 	registeredSubscriptions map[string]*api.SubscriptionMessage
 
-	stopped chan struct{}
+	pctx      context.Context
+	cancelAll context.CancelFunc
 }
 
 // New initializes and returns a new LogBroker
 func New() *LogBroker {
-	return &LogBroker{
-		logQueue:                watch.NewQueue(),
-		subscriptionQueue:       watch.NewQueue(),
-		registeredSubscriptions: make(map[string]*api.SubscriptionMessage),
-		stopped:                 make(chan struct{}),
+	return &LogBroker{}
+}
+
+// Run the log broker
+func (lb *LogBroker) Run(ctx context.Context) error {
+	lb.mu.Lock()
+
+	if lb.cancelAll != nil {
+		lb.mu.Unlock()
+		return errAlreadyRunning
+	}
+
+	lb.pctx, lb.cancelAll = context.WithCancel(ctx)
+	lb.logQueue = watch.NewQueue()
+	lb.subscriptionQueue = watch.NewQueue()
+	lb.registeredSubscriptions = make(map[string]*api.SubscriptionMessage)
+	lb.mu.Unlock()
+
+	select {
+	case <-lb.pctx.Done():
+		return lb.pctx.Err()
 	}
 }
 
 // Stop stops the log broker
-func (lb *LogBroker) Stop() {
-	close(lb.stopped)
+func (lb *LogBroker) Stop() error {
+	lb.mu.Lock()
+	defer lb.mu.Unlock()
+
+	if lb.cancelAll == nil {
+		return errNotRunning
+	}
+	lb.cancelAll()
+	lb.cancelAll = nil
+
 	lb.logQueue.Close()
 	lb.subscriptionQueue.Close()
+
+	return nil
```

Comment:
I followed the same pattern as the dispatcher and keymanager - neither of them has a blocking `Stop` (and they do interact with Raft).

Do you want me to make stop blocking or keeping it consistent with the rest?